drivers/modbus_rtu/modbus_rtu.c

```diff
@@ -0,0 +1,469 @@
+/*
+  todo: overflow buffer
+  todo: send exeption
+*/
+
+#include "modbus_rtu.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "xtimer.h"
+#include "thread.h"
+#include "periph/gpio.h"
+#include "msg.h"
+#include "checksum/ucrc16.h"
+#include "mutex.h"
+
+enum MESSAGE {
+  ID = 0,  //!< ID field
+  FUNC,    //!< Function code position
+  ADD_HI,  //!< Address high byte
+  ADD_LO,  //!< Address low byte
+  NB_HI,   //!< Number of coils or registers high byte
+  NB_LO,   //!< Number of coils or registers low byte
+  BYTE_CNT //!< byte counter
+};
+
+// bytes with crc16
+#define MIN_SIZE_REQUEST (8)
+
+#define lowByte(w) ((uint8_t)((w)&0xff))
+#define highByte(w) ((uint8_t)((w) >> 8))
```

Comment:
Maybe you can use something from the `core/include/byteorder.h` library instead?